fighters/ryu/src/status.rs

```diff
@@ -189,3 +223,111 @@ unsafe extern "C" fn ryu_attack_main_loop(fighter: &mut L2CFighterCommon) -> L2C
+        let lr =  PostureModule::lr(fighter.module_accessor);
+        if (stick_value_x.abs() > 0.66 && stick_value_y == 0.0) {
+            // airdash gfx
+            if (stick_value_x == 0.0) {
+                stick_value_x = 0.001; // to avoid a divide by zero on the next instruction below
+            }
+            let angle = (stick_value_y/stick_value_x).atan();
+            let norm_stick_pos = Vector3f { x: stick_value_x * -1.0 * lr, y: stick_value_y * -1.0, z: 0.0}.normalize();
+            let pos1 = Vector3f { x: norm_stick_pos.x * 4.0, y: 8.0 + norm_stick_pos.y * 8.0, z: 0.};
+            let pos2 = Vector3f { x: norm_stick_pos.x * 9.0, y: 8.0 + norm_stick_pos.y * 12.0, z: 0.};
+            let rot = Vector3f { x:5.0, y:0.0, z: ((stick_value_x.signum() * 90.0) + 180. * angle/3.14159)};
+            let effect_hash = Hash40::new("sys_down_smoke");
+
+            EffectModule::req_on_joint(fighter.module_accessor, effect_hash, Hash40::new("top"),
+            &pos1, &rot, 0.5, &Vector3f{x: 0.0, y: 0.0, z: 0.0}, &Vector3f{x: 0.0, y: 0.0, z: 0.0}, false, 0, 0, 0);
+
+            // transition to custom airdash status
+            let air_dash_status_kind = CustomStatusModule::get_agent_status_kind(fighter.battle_object, statuses::ryu::AIR_DASH);
+            fighter.change_status(
+                air_dash_status_kind.into(),
+                false.into()
+            );
+            return 0.into();
+        }
+    }
+    StatusModule::init_settings(
+        fighter.module_accessor,
+        app::SituationKind(*SITUATION_KIND_AIR),
+        *FIGHTER_KINETIC_TYPE_MOTION_FALL,
+        *GROUND_CORRECT_KIND_AIR as u32,
+        app::GroundCliffCheckKind(*GROUND_CLIFF_CHECK_KIND_NONE),
+        false,
+        *FIGHTER_STATUS_WORK_KEEP_FLAG_NONE_FLAG,
+        *FIGHTER_STATUS_WORK_KEEP_FLAG_NONE_INT,
+        *FIGHTER_STATUS_WORK_KEEP_FLAG_NONE_FLOAT,
+        0
+    );
+    FighterStatusModuleImpl::set_fighter_status_data(
+        fighter.module_accessor,
+        false,
+        *FIGHTER_TREADED_KIND_DISABLE,
+        false,
+        false,
+        false,
+        0,
+        0,
+        0,
+        0
+    );
+    0.into()
+}
+
+// Air Dash main status
+pub unsafe extern "C" fn air_dash_main(fighter: &mut L2CFighterCommon) -> L2CValue {
+    WorkModule::on_flag(fighter.module_accessor, *FIGHTER_STATUS_ESCAPE_AIR_FLAG_SLIDE);
+    fighter.global_table[SUB_STATUS].assign(&L2CValue::Ptr(L2CFighterCommon_sub_escape_air_uniq as *const () as _));
+    fighter.main_shift(air_dash_main_loop)
+}
+
+// Air Dash main loop
+unsafe extern "C" fn air_dash_main_loop(fighter: &mut L2CFighterCommon) -> L2CValue {
+    let frame = fighter.global_table[CURRENT_FRAME].get_i32();
+
+    if GroundModule::is_touch(fighter.module_accessor, *GROUND_TOUCH_FLAG_DOWN as u32) {
+        fighter.global_table[SITUATION_KIND].assign(&L2CValue::I32(*SITUATION_KIND_GROUND));
+    }
+
+    if frame == 6 {
```

Comment:
Since this is an integer can we do one of a few things:
1. Use `match` against these constants
2. Use ParamModule params for these so that they can be adjusted later on (for balance likely) without having to adjust the code responsible?

Reply:
This will be moved to ACMD once we implement a custom airdash anim. That's why it's formatted like ACMD for now

Reply:
I see. Please leave a `// TODO: Move to ACMD` comment in that instance